services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParserHelper.java

```diff
@@ -109,7 +110,7 @@ public String getKey() {
   }
 
   public boolean isInvalidStorageRequest() {
-    return StringUtils.isEmpty(resourceType) || (!resourceType.equals(TYPE_STORAGE))
+    return resourceType == RouterResourceType.TYPE_INVALID || resourceType != RouterResourceType.TYPE_STORAGE
```

Comment:
The first condition is redundant, isn't it? `resourceType != RouterResourceType.TYPE_STORAGE` encompasses `resourceType == RouterResourceType.TYPE_INVALID` (and many others)?

Reply:
Right! will remove it.

Reply:
actually just got rid of it.  it was a useless test only method

Reply:
Thanks! Do we have some static analysis issue?

Reply:
dont think so, locally it runs fine but fails on GH. it says some sort of timeout

Reply:
Let's figure it out...

Reply:
Weird... that means static analysis is flaky? That's concerning... was it GitHub Actions instability?

Reply:
yeah, there was one type check issue, but even after resolving that, it still failed quite a few times.